gearcmd/worker.go

```diff
@@ -32,31 +33,43 @@ func (conf TaskConfig) Process(job baseworker.Job) ([]byte, error) {
 	// This wraps the actual processing to do some logging
 	log.Println(kayvee.FormatLog("gearcmd", kayvee.Info, "START",
 		map[string]interface{}{"function": conf.FunctionName, "job_id": getJobId(job), "job_data": string(job.Data())}))
-	start := time.Now()
-	err := conf.doProcess(job)
-	end := time.Now()
-	data := map[string]interface{}{
-		"function": conf.FunctionName, "job_id": getJobId(job),
-		"job_data": string(job.Data()), "type": "gauge",
-	}
-	if err != nil {
+
+	for {
+		start := time.Now()
+		err := conf.doProcess(job)
+		end := time.Now()
+		data := map[string]interface{}{
+			"function": conf.FunctionName, "job_id": getJobId(job),
+			"job_data": string(job.Data()), "type": "gauge",
+		}
+		// Return if the job was successful.
+		if err == nil {
+			data["value"] = 1
+			data["success"] = true
+			log.Println(kayvee.FormatLog("gearcmd", kayvee.Info, "END", data))
+			// Hopefully none of our jobs last long enough for a uint64...
+			log.Printf(kayvee.FormatLog("gearcmd", kayvee.Info, "duration",
+				map[string]interface{}{
+					"value": uint64(end.Sub(start).Seconds() * 1000),
+					"type":  "gauge", "function": conf.FunctionName,
```

Comment:
Wanna take this opportunity to expand this to three lines? Same with the one above.